llvm/lib/IR/IRBuilder.cpp

```diff
@@ -1102,10 +1102,13 @@ Value *IRBuilderBase::CreateVectorSplice(Value *V1, Value *V2, int64_t Imm,
 
```

Comment:
Not directly related to this patch, but why does this function use int64_t for the immediate, but creates an i32 constant.

Reply:
I guess it's an oversight. But with the split splice.left/right intrinsics you'll be able to create splices with offsets greater than the i32 signed max. I think in a follow up patch I would propose to split  IRBuilderBase::CreateVectorSplice into IRBuilderBase::CreateVectorSpliceLeft/Right anyway.